### DISCUSSION TITLE
How to track staking rewards paid out in a specific epoch to a specific delegator of a delegation pool

### DISCUSSION THREAD

**Discussion by jcbrand** [aptos-labs/aptos-developer-discussions#188] [Questions] [answered]
### Discord user ID

defi4n+

### Describe your question in detail.

Hi, I'm looking for information on how to track staking rewards paid out to delegators of a delegated staking pool (in a specific epoch).

I see that there is a DistributeRewardsEvent in the final transaction of an epoch and this contains the pool_address and the rewards_amount.
So this tells me how many rewards were paid out to the pool, but how can I find out how many rewards accrue to a specific delegator to that pool?

Thanks 🙏

### What have you tried or looked at?  Or how can we reproduce the error?

I've been looking at transactions with reward payouts in an explorer.

For example:
https://explorer.aptoslabs.com/txn/590484333/events?network=mainnet

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

N/A

**Comment by 0x-j** [comment, marked as answer]
I found this [doc](https://aptos.dev/nodes/validator-node/operator/delegation-pool-operations/#compute-delegation-pool-rewards-earned), it might give you some ideas. 

I don't have a good solution right now, and i'm asking around.

This [thread](https://github.com/aptos-labs/aptos-developer-discussions/discussions/114#discussioncomment-8967472) might also be helpful.

A pretty complex solution I can think of.
```ts
import { Aptos, AptosConfig, Network } from "@aptos-labs/ts-sdk";

const config = new AptosConfig({
  network: Network.MAINNET,
});
export const aptos = new Aptos(config);

export const calculateReward = async () =>
  {
    const delegatorAddr =
      "0x42b7cd08d63005d04c8dd0ed5e7e1acab3cf20bf77dab114ba5a2806d0b97be2";
    const poolAddress =
      "0x9bfd93ebaa1efd65515642942a607eeca53a0188c04c21ced646d2f0b9f551e8";

    const [activeStake, inactiveStake, pendingInactiveStake] = await aptos.view(
      {
        payload: {
          function: "0x1::delegation_pool::get_stake",
          typeArguments: [],
          functionArguments: [poolAddress, delegatorAddr],
        },
        options: {
          ledgerVersion: 0, // last ledger version of the epoch
        },
      }
    );

    // find how much APT users staked during each epoch. subtraction gives you reward at each epoch
  };
```

**Reply by buzzlamicjihad** [reply]
This was helpful, thank you

**Reply by jcbrand** [reply]
Thanks @0xaptosj  :pray:

I would have preferred a solution that doesn't require us to keep track of how much we've staked per epoch, but looks like it's not possible ATM.

**Reply by 0x-j** [reply]
Yeah unfortunately that's not supported now 😢
